pyk/src/pyk/__main__.py

```diff
@@ -389,6 +395,65 @@ def exec_json_to_kore(options: JsonToKoreOptions) -> None:
     sys.stdout.write('\n')
 
 
+def exec_parse_outer(options: ParseOuterOptions) -> None:
```

Comment:
Please extract a function into one of the `pyk.kast` modules that performs the transformation logic. Unit tests can call that function directly.

Reply:
Sorry, what transformation logic are you referring to?

Reply:
This function performs side effects. You can extract function that returns `result_text` (or something more structured), that way, the command line handler becomes very simple.

Reply:
Sounds good.

You've already set up a test (`TestParseOuter`) though, what I'm saying, if you extract a function (that takes `main_file`, etc. and returns the text you'd otherwise print), then you can call that directly without stubbing `sys.argv`.

Reply:
Yes, I agree that the substantive work here should be extracted into a function that _returns_ something - the implementation will look pretty similar to what you have here, just with a different signature